docs/ref/models/expressions.txt

```diff
@@ -127,6 +127,21 @@ Although ``reporter.stories_filed = F('stories_filed') + 1`` looks like a
 normal Python assignment of value to an instance attribute, in fact it's an SQL
 construct describing an operation on the database.
 
+For fields that are string-expressions
+(or ``django.contrib.postgres.fields.ArrayField``), you can use the standard
+slicing notation. The indices used in slicing are 0-based. The step-argument is
+not supported. For example::
+
+    >>> # Replacing a name with its substring
+    >>> writer = Writers.objects.get(name='Priyansh')
+    >>> writer.name = F('name')[1:5]
+    >>> writer.save()
+    >>> writer.refresh_from_db()
+    >>> writer.name
+    'riya'
+
+.. versionadded:: 4.0
```

Comment:
Please move this to the bottom of the section and add some content, for example:

```suggestion
.. versionadded:: 4.0

    Support for slicing the field was added where supported by the data type.
```

This allows it to be removed easily when these annotations are dropped after a few releases.